addon/utils/store.js

```diff
@@ -21,11 +21,30 @@ export default EmberObject.extend({
   },
 
   initialsFor(properties) {
-    let key = hash(properties);
+    let hashableProp = this._ensureHashable(properties);
+    let key = md5(hashableProp);
     return this.get('cache')[key] || this._create(key, properties);
   },
 
   _create(key, properties) {
     return this.get('cache')[key] = this.get('generator').generate(properties);
+  },
+
+  _ensureHashable(properties) {
+    if (typeOf(properties) !== 'object') {
+      // this should never happen, but just in case
```

Comment:
Actually, it can happen because the value of the properties key can be a whatever

Reply:
👍 I'll add some test cases to cover it.